src/openforms/emails/tests/test_digest_functions.py

```diff
@@ -0,0 +1,407 @@
+            status=Message.STATUS_SENT,
+            status_label="Sent",
+            include_in_daily_digest=True,
+        )
+
+        failed_emails = collect_failed_emails(
+            since=datetime(2023, 2, 1, 0, 0, 0).replace(tzinfo=utc)
+        )
+
+        self.assertEqual(len(failed_emails), 1)
+        self.assertEqual(failed_emails[0].submission_uuid, submission.uuid)
+
+    def test_sent_emails_are_not_collected(self):
+        submission = SubmissionFactory.create()
+
+        # Log email failed more than 24h ago
+        logevent.email_status_change(
+            submission,
+            event="registration",
+            status=Message.STATUS_FAILED,
+            status_label="Failed",
+            include_in_daily_digest=True,
+        )
+
+        failed_emails = collect_failed_emails(
+            since=datetime(2023, 2, 2, 13, 30, 0).replace(tzinfo=utc)
+        )
+
+        self.assertEqual(failed_emails, [])
+
+
+@freeze_time("2023-01-02T12:30:00+01:00")
+class FailedRegistrationsTests(TestCase):
+
+    def test_failed_registrations_are_collected(self):
+        # 1st form with 2 failures in the past 24 hours
+        form_1 = FormFactory.create()
+        failed_submission_1 = SubmissionFactory.create(
+            form=form_1, registration_status=RegistrationStatuses.failed
+        )
+        failed_submission_2 = SubmissionFactory.create(
+            form=form_1, registration_status=RegistrationStatuses.failed
+        )
+
+        # 1st failure
+        logevent.registration_failure(
+            failed_submission_1,
+            RegistrationFailed("Registration plugin is not enabled"),
+        )
+
+        # 2nd failure
+        logevent.registration_failure(
+            failed_submission_2,
+            RegistrationFailed("Registration plugin is not enabled"),
+        )
+
+        # 2nd form with 1 failure in the past 24 hours
+        form_2 = FormFactory.create()
+        failed_submission = SubmissionFactory.create(
+            form=form_2, registration_status=RegistrationStatuses.failed
+        )
+
+        # failure
+        logevent.registration_failure(
+            failed_submission,
+            RegistrationFailed("Registration plugin is not enabled"),
+        )
+
+        failed_registrations = collect_failed_registrations(
+            since=datetime(2023, 1, 1, 14, 30, 0).replace(tzinfo=utc)
+        )
+
+        self.assertEqual(len(failed_registrations), 2)
+        self.assertEqual(failed_registrations[0].failed_submissions_counter, 2)
+        self.assertEqual(failed_registrations[1].failed_submissions_counter, 1)
+
+    def test_successful_registrations_are_not_collected(self):
+        form = FormFactory.create()
+        SubmissionFactory.create(form=form, registration_success=True)
+
+        failed_registrations = collect_failed_registrations(
+            since=datetime(2023, 1, 2, 14, 30, 0).replace(tzinfo=utc)
+        )
+
+        self.assertEqual(failed_registrations, [])
```

Comment:
I don't think any log events are created here for succesful registration, so this test would always pass.